tests/compile-fail/invalid_bool.rs

```diff
@@ -2,7 +2,9 @@
 // Make sure we find these even with many checks disabled.
 // compile-flags: -Zmiri-disable-alignment-check -Zmiri-disable-stacked-borrows -Zmiri-disable-validation
 
+fn dont_optimize<T>(x: T) -> T { x }
```

Comment:
```suggestion
#[inline(never)]
fn dont_optimize<T>(x: T) -> T { x }
```

The mir inliner may be able to inline it in the future.